homeassistant/backup_restore.py

```diff
@@ -24,15 +25,27 @@ class RestoreBackupFileContent:
     """Definition for restore backup file content."""
 
     backup_file_path: Path
+    password: str | None = None
+
+
+def password_to_key(password: str | None) -> bytes | None:
+    """Generate a AES Key from password."""
+    if password is None:
```

Comment:
I suggest we remove this case and make password required in this function. The caller should only use this function when there's a password.